config/webhook/manifests.yaml

```diff
@@ -0,0 +1,29 @@
+
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  creationTimestamp: null
+  name: validating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1
+  - v1beta1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
```

Comment:
This cannot be changed. However that's not a problem because the namespace is actually injected by kustomize using the configuration:
https://github.com/openshift/external-dns-operator/blob/main/config/crd/kustomizeconfig.yaml

Reply:
Didn't quite get how CRD's kustomizeconfig changes the clientConfig in ValidatingWebHook.
The namespace in the clientConfig of the webhook seems to come from [this configuration](https://github.com/openshift/external-dns-operator/blob/666d26cc7b7a04cf565068bf23fa4ef4541ae271/config/default/kustomization.yaml#L2) somehow.

Service namespace is `andrey` but webhook's client config is still `system` because no global namespace is given:
```bash
$ git diff -U0
--- a/config/default/kustomization.yaml
+++ b/config/default/kustomization.yaml
@@ -2 +2 @@
-namespace: external-dns-operator
+#namespace: external-dns-operator

--- a/config/webhook/service.yaml
+++ b/config/webhook/service.yaml
@@ -6 +6 @@ metadata:
-  namespace: external-dns-operator
+  namespace: andrey

$ oc apply -k config/default/ --dry-run=client -o yaml | grep -A4 clientConfig:
    clientConfig:
      service:
        name: webhook-service
        namespace: system
        path: /validate-externaldns-olm-openshift-io-v1alpha1-externaldns
```

But when the global namespace is there, it's used:
```bash
$ git diff -U0
--- a/config/default/kustomization.yaml
+++ b/config/default/kustomization.yaml
@@ -2 +2 @@
-namespace: external-dns-operator
+namespace: external-dns-operator-andrey

$ $ oc apply -k config/default/ --dry-run=client -o yaml | grep -A4 clientConfig:
    clientConfig:
      service:
        name: webhook-service
        namespace: external-dns-operator-andrey
        path: /validate-externaldns-olm-openshift-io-v1alpha1-externaldns
```

Reply:
Oouf, I think I got it. It's so called custom transformer which can modify the meaning of the transformation.
In case of the webhook's kustomize directory the custom transformer was used to "redirect" the namespace definition to a different place, not the standard `metadata.namespace` but [here](https://github.com/openshift/external-dns-operator/blob/666d26cc7b7a04cf565068bf23fa4ef4541ae271/config/webhook/kustomizeconfig.yaml#L21). So, the namespace specified in `config/default` was added to the custom path for the validating webhook.

Anyway, you are right, we can keep it set to `system`.